AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2664,6 +2668,67 @@ - (void)_enqueueAsyncSizingWithSentinel:(ASSentinel *)sentinel completion:(void(
   return self;
 }
 
+#pragma mark - ASEnvironment
+
+- (ASEnvironmentCollection *)environmentCollection
+{
+  return &_environmentCollection;
```

Comment:
For the next round of work on this, top priority is to make these kind of accesses threadsafe.  The only way I know how to do that is ensure we never return a pointer to the structure (including _any_ sub-structure of the environment), or there isn't a way to lock access to it.